tests/functional/glusterd/test_glusterd_logs_when_peer_detach.py

```
@@ -0,0 +1,65 @@
+  with this program; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+  Description:
+        No Errors should generate in glusterd.log while detaching
+        node from gluster
+"""
+
+import random
+from tests.abstract_test import AbstractTest
+
+# disruptive;
+
+
+class GlusterdLogsWhilePeerDetach(AbstractTest):
+
+    def run_test(self, redant):
+        '''
+            1) Detach the node from peer
+            2) Check that any error messages related to peer detach
+                in glusterd log file
+            3) No errors should be there in glusterd log file
+        '''
+
+        # Getting timestamp
+        ret = redant.execute_abstract_op_node('date +%s', self.server_list[0])
+        timestamp = ret['msg'][0].rstrip("\n")
+
+        #  glusterd logs
+        ret = redant.execute_abstract_op_node('cp /var/log/glusterfs/'
+                                              'glusterd.log /var/log/'
+                                              'glusterfs/'
+                                              f'glusterd_{timestamp}.log',
+                                              self.server_list[0])
+
+        # Clearing the existing glusterd log file
+        ret = redant.execute_abstract_op_node('echo > /var/log/glusterfs/'
+                                              'glusterd.log',
+                                              self.server_list[0])
+
+        # Performing peer detach
+        random_server = random.choice(self.server_list[1:])
+        ret = redant.peer_detach(random_server, self.server_list[0])
+
+        # Searching for error message in log
+        ret = redant.execute_abstract_op_node("grep ' E ' /var/log/glusterfs/"
+                                              "glusterd.log | wc -l",
+                                              self.server_list[0])
+        if int(ret['msg'][0]) != 0:
+            redant.logger.info("Found Error messages in glusterd log "
```

We should raise an exception here rather than using the `redant.logger.error`, we should be doing something like
```        if int(ret['msg'][0]) != 0:
                 raise Exception("Found Error messages in glusterd log file after peer detach")```

---

Ack